dbt/adapters/redshift/relation_configs/sort.py

```diff
@@ -147,20 +148,24 @@ def parse_relation_results(cls, relation_results_entry: "agate.Row") -> dict:
             Processing of `sortstyle` has been omitted here, which means it's the default (compound).
 
         Args:
-            relation_results_entry: the description of the sortkey and sortstyle from the database in this format:
-
-                agate.Row({
-                    ...,
-                    "sortkey1": "<column_name>",
-                    ...
-                })
+            relation_results_entry: The list of rows that contains the sortkey in this format:
+                [
+                    agate.Row({
+                        ...,
+                        "column": "<column_name>",
+                        "is_sort_key": True,
+                        ...
+                    }),
+                ]
 
         Returns: a standard dictionary describing this `RedshiftSortConfig` instance
         """
-        if sortkey := relation_results_entry.get("sortkey1"):
-            return {"sortkey": {sortkey}}
-        return {}
+        sort_config = []
+        for column in relation_results_entry:
```

Comment:
Per the comment above, we should make sure this config contains the sort key columns in the correct sort order.

Reply:
fixed